partcad/src/partcad/__init__.py

```
@@ -84,17 +91,77 @@ def before_send(event, hint):
+
+    username = os.getenv("USER")
+    username_md5 = hashlib.md5(username.encode()).hexdigest()
+    sentry_sdk.set_tag("username.md5", username_md5)
+
+    cache_dir = os.path.expanduser("~/.cache/partcad")
+    os.makedirs(cache_dir, exist_ok=True)
+    cache_file = os.path.join(cache_dir, "id")
+
+    # Check if the ID is already cached
+    if os.path.exists(cache_file):
+        with open(cache_file, "r") as f:
```

Ruff (0.8.2)

Unnecessary open mode parameters

Remove open mode parameters

(UP015)